tests/org.eclipse.ui.workbench.texteditor.tests/src/org/eclipse/ui/findandreplace/TestFindReplaceLogic.java

```diff
@@ -0,0 +1,275 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Vector Informatik GmbH and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Vector Informatik GmbH - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.ui.findandreplace;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.After;
+import org.junit.Test;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Shell;
+
+import org.eclipse.jface.text.Document;
+import org.eclipse.jface.text.TextViewer;
+
+public class TestFindReplaceLogic {
+
+	Shell parentShell;
+	IFindReplaceLogic findReplaceLogic;
+	TextViewer textViewer;
+
+	public void setupFindReplaceLogicObject() {
+		setupFindReplaceLogicObject(null);
+	}
+
+	public void setupFindReplaceLogicObject(TextViewer target) {
+		findReplaceLogic= new FindReplaceLogic();
+		if (target != null) {
+			findReplaceLogic.updateTarget(target.getFindReplaceTarget(), true);
+		}
+	}
+
+	public void setupTextViewer(String contentText) {
+		parentShell= new Shell();
+		textViewer= new TextViewer(parentShell, SWT.BORDER | SWT.V_SCROLL | SWT.H_SCROLL);
+		textViewer.setDocument(new Document(contentText));
+		textViewer.getControl().setFocus();
+	}
+
+	@After
+	public void tearDownObjects() {
+		if (textViewer != null) {
+			textViewer.getControl().dispose();
+			textViewer= null;
+		}
+		parentShell.dispose();
+	}
+
+	@SuppressWarnings("boxing")
+	private void expectStatusIs(FindReplaceLogicMessageStatus test, FindReplaceLogicMessageStatus expected) {
+		assertEquals("Status message is not the expected message", test.getMessage(), expected.getMessage());
+		assertEquals("Status warning flag is not the expected value", test.isWarning(), expected.isWarning());
+		assertEquals("Status error flag is not the expected value", test.isError(), expected.isError());
+		findReplaceLogic.resetStatus(); // sketchy side-effect of this function.
+	}
+
+	private final String baseReplaceAllTestCaseSetupString= "aaaa";
+	/**
+	 * Expects the TextViewer to contain a document with "aaaa"
+	 */
+	private void performReplaceAllBaseTestcases() {
+		Display display= parentShell.getDisplay();
+
+		findReplaceLogic.performReplaceAll("a", "b", display);
+		assertEquals("Text was not correctly replaced", "bbbb", textViewer.getDocument().get());
+		expectStatusIs(findReplaceLogic.getStatus(), new FindReplaceLogicMessageStatus("4 matches replaced", false, false));
```

Comment:
It is not a good idea to check the code for using specific strings, as they may change, may be localized etc.

Probably this shows that it is not a good idea to have a `FindReplaceLogicMessageStatus` that contains a string generated by the logic (as that is actually UI functionality). Probably the status object should contains the actualy status (number of matches etc.) and leave it up to the UI to interpret that information, e.g., in terms of an according string.